### PR TITLE
feat(842): Add route for getting template tags

### PR DESCRIPTION
**Pull request by ScottyFillups**
## Context

It's hard for users to figure out what tags are available for each template and what versions they correspond to. It would be nice to have an endpoint where we could list the available tags for a specific template name.

## Objective

Create an endpoint, GET `/templates/{name}/tags`, which returns template tags.

### CODE REVIEW

File: plugins/templates/listTags.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const listSchema = joi.array().items(schema.models.templateTag.base).label('List of templates');
+const nameSchema = joi.reach(schema.models.templateTag.base, 'name');
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/templates/{name}/tags',
+    config: {
+        description: 'Get all template tags for a given template name with pagination',
```
Comment (d2lam) [inline plugins/templates/listTags.js]: We actually don't have pagination right now: https://github.com/screwdriver-cd/screwdriver/issues/233

File: plugins/templates/listTags.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        notes: 'Returns all template records for a given template name',
```
Comment (d2lam) [inline plugins/templates/listTags.js]: This one returns the template tags only, not the whole template records right?

File: test/plugins/data/templateTags.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[{
+    "id": 7969,
+    "name": "screwdriver/build",
+    "tag": "stable",
+    "version": "0.0.1"
+},
+{
+    "id": 7970,
+    "name": "screwdriver/build",
+    "tag": "stable",
```
Comment (d2lam) [inline test/plugins/data/templateTags.json]: Hmm to be accurate, 1 tag can only be mapped to 1 version. There cannot be multiple versions with the `stable` tag. For example, `stable` can only map to `3.2.1`. 
However, 1 version can have multiple tags. For example, `3.2.1` can be `stable` and `latest` at the same time. It's very similar to docker hub tags.

File: plugins/templates/listTags.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            return factory.list({
+                params: { name: request.params.name },
+                paginate: {
+                    page: request.query.page,
+                    count: request.query.count
+                },
+                sort: request.query.sort
+            }).then((templates) => {
+                if (templates.length === 0) {
+                    throw boom.notFound('Template does not exist');
```
Comment (minz1027) [inline plugins/templates/listTags.js]: It's not templates, it's template tags. A template can have zero tags.

File: plugins/templates/listTags.js (hunk 1 above)
Comment (d2lam) [reply plugins/templates/listTags.js]: @ScottyFillups  We should remove the `with pagination`

File: plugins/templates/listTags.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            const factory = request.server.app.templateTagFactory;
+
+            return factory.list({
+                params: { name: request.params.name },
+                paginate: {
+                    page: request.query.page,
+                    count: request.query.count
+                },
+                sort: request.query.sort
+            }).then((templates) => {
```
Comment (minz1027) [inline plugins/templates/listTags.js]: I will prefer we use `tags` instead of `templates` everywhere

**Pull request merged**